src/store/auth.js

```diff
@@ -0,0 +1,43 @@
+import api from '../plugins/axios';
+import { saveToken } from '../common/jwt.service';
+
+const state = () => ({
+  curUser: {},
+  isAuthenticated: false
+});
+
+const getters = {
+  curUser: state => state.curUser,
+  isAuth: state => state.isAuthenticated
+};
+
+const mutations = {
+  setAuthState: (state, isAuth) => (state.isAuthenticated = isAuth)
+};
+
+const actions = {
+  // sign in user and save token
+  signin({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
+        saveToken(res.data.token);
+        commit('setAuthState', true);
+      })
+      .catch(error => console.log('Sign in ERROR --- ' + error));
+  },
```

Comment:
1/ Should create service file to handle call API.
 - Create folder services and then create file {module}.js to handle call API. Example services/book.js, services/auth.js
 - And in action, import service to use
2/ Should try async/await

Reply:
Done